mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -24,11 +25,55 @@ def SYCL_Dialect : Dialect {
   let name = "sycl";
   let cppNamespace = "::mlir::sycl";
   let useDefaultTypePrinterParser = 1;
+  let extraClassDeclaration = [{
+    llvm::DenseMap<std::pair<::mlir::TypeID, llvm::StringRef>, llvm::StringRef>
+        Methods;
+
+    static llvm::StringRef getBaseTypeAttrName() { return "BaseType"; }
+
+    static llvm::StringRef getBaseTypeNameAttrName() { return "BaseTypeName"; }
+
+    static llvm::StringRef getFunctionNameAttrName() { return "Function"; }
+
+    static llvm::StringRef getMangledFunctionNameAttrName() { return "MangledName"; }
+
+    void addSYCLMethod(
```

Comment:
Will this function add a SYCL method given its class type ID, its unmangled name and the does it add the method into the MethodNames ?  Adding a comment would help quickly understand the functionality provided here.